arangod/StorageEngine/VPackSortMigration.cpp

```diff
@@ -0,0 +1,284 @@
+                          << " and name " << index->name();
+                      alarm = true;
+                      break;
+                    }
+                  }
+                  prev_key.assign(key.data(), key.size());
+                }
+                if (alarm) {
+                  problemFound = true;
+                  {
+                    VPackObjectBuilder guardO2(&result);
+                    guardO2->add("database", VPackValue(name));
+                    guardO2->add("collection", VPackValue(collection->name()));
+                    guardO2->add("indexId", VPackValue(index->id().id()));
+                    guardO2->add("indexName", VPackValue(index->name()));
+                  }
+                }
+              }
+            }
+          }
+        });
+      }
+    }
+
+    if (!problemFound) {
+      guardO->add(StaticStrings::Error, VPackValue(false));
+      guardO->add(StaticStrings::ErrorCode, VPackValue(0));
+      guardO->add(StaticStrings::ErrorMessage,
+                  VPackValue("all good with sorting order"));
+    } else {
+      guardO->add(StaticStrings::Error, VPackValue(true));
+      guardO->add(StaticStrings::ErrorCode,
+                  VPackValue(TRI_ERROR_ARANGO_INDEX_HAS_LEGACY_SORTED_KEYS));
+      guardO->add(StaticStrings::ErrorMessage,
+                  VPackValue("some indexes have legacy sorted keys"));
+    }
+  }
+
+  return {};
+}
+
+Result migrateVPackIndexSorting(TRI_vocbase_t& vocbase, VPackBuilder& result) {
+  // Just for the sake of completeness, restrict ourselves to the RocksDB
+  // storage engine:
+  result.clear();
+  auto& engineSelectorFeature =
+      vocbase.server().getFeature<EngineSelectorFeature>();
+  if (!engineSelectorFeature.isRocksDB()) {
+    return Result(TRI_ERROR_NOT_IMPLEMENTED,
+                  "VPack sorting migration is unnecessary for storage engines "
+                  "other than RocksDB");
+  }
+
+  auto& engine = engineSelectorFeature.engine<RocksDBEngine>();
+  Result res = engine.writeSortingFile(
+      arangodb::basics::VelocyPackHelper::SortingMethod::Correct);
+
+  if (res.ok()) {
+    VPackObjectBuilder guard(&result);
+    guard->add(StaticStrings::Error, VPackValue(false));
+    guard->add(StaticStrings::ErrorCode, VPackValue(0));
+    guard->add(StaticStrings::ErrorMessage,
+               VPackValue("VPack sorting migration done."));
+    return {};
+  }
+  return res;
+}
+
+namespace {
+Result fanOutRequests(TRI_vocbase_t& vocbase, fuerte::RestVerb verb,
+                      VPackBuilder& result, bool includeAgents) {
+  TRI_ASSERT(ServerState::instance()->isCoordinator());
+
+  auto& ci = vocbase.server().getFeature<ClusterFeature>().clusterInfo();
+  auto& nf = vocbase.server().getFeature<NetworkFeature>();
+
+  auto dbs = ci.getCurrentDBServers();
+  std::vector<futures::Future<network::Response>> requests;
+  requests.reserve(dbs.size());
+
+  network::RequestOptions opts;
+  opts.database = vocbase.name();
```

Comment:
By the way, do we want to increase the default timeout here?
The default timeout used by the NetworkFeature is 120.0s, which may not be enough to iterate over all vpack indexes of a DB server/single instance without running into a request timeout.
(this also applies to the devel version of the changes)